digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ProhibitionService.scala

```diff
@@ -0,0 +1,191 @@
+package fi.liikennevirasto.digiroad2
+
+import fi.liikennevirasto.digiroad2.asset.{SideCode, UnknownLinkType}
+import fi.liikennevirasto.digiroad2.linearasset._
+import fi.liikennevirasto.digiroad2.linearasset.oracle.OracleLinearAssetDao
+import fi.liikennevirasto.digiroad2.util.{LinearAssetUtils, PolygonTools}
+import org.joda.time.DateTime
+import slick.jdbc.StaticQuery.interpolation
+import slick.driver.JdbcDriver.backend.Database.dynamicSession
+
+class ProhibitionService(roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus) extends LinearAssetOperations {
+  override def roadLinkService: RoadLinkService = roadLinkServiceImpl
+  override def dao: OracleLinearAssetDao = new OracleLinearAssetDao(roadLinkServiceImpl.vvhClient, roadLinkServiceImpl)
+  override def eventBus: DigiroadEventBus = eventBusImpl
+  override def vvhClient: VVHClient = roadLinkServiceImpl.vvhClient
+  override def polygonTools: PolygonTools = new PolygonTools()
+
+  override def getUncheckedLinearAssets(areas: Option[Set[Int]]) = throw new UnsupportedOperationException("Not supported method")
+
+
+  /**
+    * Returns linear assets by asset type and asset ids. Used by Digiroad2Api /linearassets POST and /linearassets DELETE endpoints.
+    */
+  override def getPersistedAssetsByIds(typeId: Int, ids: Set[Long]): Seq[PersistedLinearAsset] = {
+    withDynTransaction {
+      dao.fetchProhibitionsByIds(typeId, ids)
+    }
+  }
+
+  override def getAssetsByMunicipality(typeId: Int, municipality: Int): Seq[PersistedLinearAsset] = {
+    val (roadLinks, changes) = roadLinkService.getRoadLinksWithComplementaryAndChangesFromVVH(municipality)
+    val roadLink: Seq[RoadLink] = roadLinks.filter(_.functionalClass > 4 || typeId != LinearAssetTypes.MaintenanceRoadAssetTypeId)
```

Comment:
Remove this filtering because is to be used by the MaintenanceService and not by the ProhibitionService